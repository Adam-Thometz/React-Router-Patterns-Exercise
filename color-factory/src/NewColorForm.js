import React, {useState} from "react";
import { Link, Redirect } from "react-router-dom";

const NewColorForm = () => {
  const init = {
    name: "",
    color: ""
  }

  const [formData, setFormData] = useState(init)
  const [submitted, setSubmitted] = useState(false)

  const handleChange = e => {
    const {name, value} = e.target
    setFormData(data => ({
      ...data,
      [name]: value
    }))
  }

  const toggleSubmitButton = () => {
    // If everything in the form is filled out, set disabled attribute on submit button to false
    if (Object.values(formData).every(data => data !== '')) {
      return false
    } else {
      return true
    }
  }

  const handleSubmit = e => {
    e.preventDefault()
    const {name, color} = formData
    localStorage.setItem(name, color)
    setSubmitted(true)
  }

  return (
    <form className="NewColorForm">
      <label htmlFor="name">Color Name: </label>
      <input
        type="text"
        id="name"
        name="name"
        value={formData.name}
        onChange={handleChange}
      />
      <label htmlFor="color">Color Value: </label>
      <input
        type="color"
        id="color"
        name="color"
        value={formData.color}
        onChange={handleChange}
      />
      <button disabled={toggleSubmitButton()} onClick={handleSubmit}>Add color!</button>
      <Link to="/colors">Go back</Link>
      {submitted ? <Redirect to="/colors" /> : null}
    </form>
  )
}

export default NewColorForm