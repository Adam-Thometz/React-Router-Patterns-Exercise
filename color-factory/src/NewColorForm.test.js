import { render, screen } from '@testing-library/react';
import NewColorForm from './NewColorForm';
import {MemoryRouter} from 'react-router-dom'

test('renders component', () => {
  render(<MemoryRouter>
    <NewColorForm />
  </MemoryRouter>);
});

test('matches snapshot', () => {
  const {asFragment} = render(<MemoryRouter>
    <NewColorForm />
  </MemoryRouter>)
  expect(asFragment()).toMatchSnapshot()
})