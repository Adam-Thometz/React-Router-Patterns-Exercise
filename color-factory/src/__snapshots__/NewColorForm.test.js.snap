// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="NewColorForm"
  >
    <label
      for="name"
    >
      Color Name: 
    </label>
    <input
      id="name"
      name="name"
      type="text"
      value=""
    />
    <label
      for="color"
    >
      Color Value: 
    </label>
    <input
      id="color"
      name="color"
      type="color"
      value=""
    />
    <button
      disabled=""
    >
      Add color!
    </button>
    <a
      href="/colors"
    >
      Go back
    </a>
  </form>
</DocumentFragment>
`;
